lib/locations/reader.rb

```diff
@@ -0,0 +1,45 @@
+require 'open-uri'
+
+module Locations
+  class Reader
+    attr_accessor :path, :redis
+
+    def initialize(path, redis = Redis.new(uri: redis_uri))
+      self.path = path
+      self.redis = redis
+    end
+
+    def call
+      if json_from_url?
+        cached_json
+      else
+        read_json(path)
+      end
+    end
+
+    private
+
+    def json_from_url?
+      path.to_s =~ /^http/
+    end
+
+    def cached_json
+      json = redis.get('locations')
+      unless json
+        json = read_json(path)
+        redis.setex('locations', 60 * 60, json)
```

Comment:
The Go app isn't setting any headers currently and I'm not sure if we're using CloudFlare for this either.

The net effect of this code is to merely reduce the need for an outgoing HTTP request every time the repository is instantiated.

As has been touched on briefly, this could be extended into a cron job which pulls down the locations GeoJSON every hour which would allow the guidance app to just look in Redis for the data.